### ISSUE TITLE
username contains invalid characters

### ISSUE THREAD

**Issue by hibobmaster** [keycloak/keycloak#28231]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/api

### Describe the bug

After upgrade to Keycloak v24.0.2，i create a user and its username with CJK characters report 'Username' contains invalid character.

From admin panel:
![image](https://github.com/keycloak/keycloak/assets/32976627/6ae4b354-3e51-44eb-aad8-0bb71a88a0ff)

From API:
{"field":"username","errorMessage":"error-username-invalid-character","params":["username"]}

### Version

24.0.2

### Regression

- [X] The issue is a regression

### Expected behavior

Username can contain CJK characters.

### Actual behavior

I cannot register user with a username containing CJK characters.

### How to Reproduce?

Create a user with a username containing CJK characters, such as "测试12345".

**Comment by abstractj** [comment]
@keycloak/ui-maintainers could you please double-check? It seems, the issue might not be related to the endpoint after reviewing the HTTP request.

**Comment by jonkoops** [comment]
Seems pretty clear cut to me, the API considers this invalid.

**Comment by mposolda** [comment]
@hibobmaster This is because of user-profile is enabled by default in Keycloak 24 and there is default validation on `username` attribute, which prohibits username like `测试12345` by default. This is documented here: https://www.keycloak.org/docs/latest/upgrading/index.html#default-validations .

Could you please doublecheck if removing default validation of username (I think especially `username-prohibited-characters` in this case) fixes the issue for you?

**Comment by hibobmaster** [comment]
> @hibobmaster This is because of user-profile is enabled by default in Keycloak 24 and there is default validation on `username` attribute, which prohibits username like `测试12345` by default. This is documented here: https://www.keycloak.org/docs/latest/upgrading/index.html#default-validations .
> 
> Could you please doublecheck if removing default validation of username (I think especially `username-prohibited-characters` in this case) fixes the issue for you?

Thanks for your response. After removing `username-prohibited-characters` validation and save the change, the problem still exists, even after restarting the keycloak instance.
![Snipaste_2024-04-02_20-52-31](https://github.com/keycloak/keycloak/assets/32976627/5196c740-1c1e-4cc7-9a3b-81aa986f3849)

![image](https://github.com/keycloak/keycloak/assets/32976627/9ad2f0d4-b3e5-4d50-935c-fc27d7eaace3)

**Comment by hibobmaster** [comment]
Ok, it works after removing `up-username-not-idn-homograph` validation.

**Comment by sguilhen** [comment]
@hibobmaster Can we close this one or do you think there's still something that needs to be looked at?

**Comment by hibobmaster** [comment]
@sguilhen It's okay now, thank you for all of your help. Besides, i would suggest mention it in document.

**Comment by Amrooo161** [comment]
Is there a way to not make keycloak create those validators by default, even when importing an old realm-json file the realm gets created with those validators.

I tried supplying an empty config for them but keycloak still created those, I don't think they should be created by default not everyone who uses keycloak would need them.

**Comment by kherock** [comment]
Yes I had this resolved in Keycloak 22 by removing the `up-username-not-idn-homograph`. However, in Keycloak 24, it seems like this is no longer possible. Why is removing the built-in validators not allowed?
